src/Mir/Trans.hs

```diff
@@ -687,11 +688,11 @@ evalCast' ck ty1 e ty2  =
       (M.Misc, TyBool, TyUint bsz)
        | MirExp C.BoolRepr e0 <- e
        -> baseSizeToNatCont bsz $ \w -> 
-           return $ MirExp (C.BVRepr w) (R.App $ E.BVIte e0 w (R.App $ E.BVLit w 1) (R.App $ E.BVLit w 0))
+           return $ MirExp (C.BVRepr w) (R.App $ E.BVIte e0 w (R.App $ E.BVLit w (BV.mkBV w 1)) (R.App $ E.BVLit w (BV.mkBV w 0)))
```

Comment:
This pattern `BVLit w (mkBV w x)` shows up a lot here - would it make sense to define a common helper function?  I assume the same pattern would show up fairly often in other Crucible client code as well.

Reply:
Yes, we could certainly introduce a small combinator to wrap that up.  We didn't do it in the other codebases to make the diffs as direct as possible for easier inspection.  I can introduce one here.